File: Creational/Lab4/Python/estate_to_rent_factory.py
from estate_factory import *

class EstateToRentFactory(EstateFactory):
    def create_estate(self, type: str) -> RealEstate:
        e = self.estate_from_type(type)
        e.is_buyable = False
        e.is_rentable = True
        e.is_sellable = False
        return e

File: Creational/Lab4/C#/Factories/EstateToRentFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_.Factories
{
    public class EstateToRentFactory : EstateFactory
    {
        public override RealEstate CreateEstate(string type)
        {
            var e = EstateFromType(type);
            e.IsBuyable = false;
            e.IsRentable = true;
            e.IsSellable = false;
            return e;
        }
    }
}
